ClassLibrary/ClassLibrary/Library.cs

```
@@ -1,13 +1,51 @@
 ﻿using System;
 using System.Collections.Generic;
+using System.Linq;
 
 namespace ClassLibrary
 {
     public class Library
     {
+        //List All Books
         public List<Book> Books= new List<Book>();
-        public List<CardReader> CardReaders = new List<CardReader>();
-        public List<Reader> Readers = new List<Reader>();
+
+        //List All Athor
+        public List<Author> Authors = new List<Author>();
+        //public List<CardReader> CardReaders = new List<CardReader>();
+        //public List<Reader> Readers = new List<Reader>();
+
+        //букк пока не смотри
+        public void AddBook(String ISBN, String NameBook,String FullNameAuthor)
+        {
+            Book newBook = new Book();
+        }
+
+        public Author AddAuthor(String FullNameAuthor)
+        {
+            if (FullNameAuthor.IsNullOrWhiteSpace()) throw new NullReferenceException("Имя автора пусто");
+            String[] fullNameAuthor = FullNameAuthor.Split(' ');
+            if (fullNameAuthor?.Length!=2) throw new ArgumentException("Неправильное имя автора");
+
+            return AddAuthor(fullNameAuthor[0], fullNameAuthor[0]);
+        }
+
+        public Author AddAuthor(String nameAuthor,String lastNameAuthor)
+        {
+            if (nameAuthor.IsNullOrWhiteSpace()|| lastNameAuthor.IsNullOrWhiteSpace()) throw new NullReferenceException("Имя или фамилии автора пусто");
+            Author authorInList = Authors?.FirstOrDefault(author =>
+                author.Name == nameAuthor && author.LastName == lastNameAuthor);
+
+            Author returnAuthor;
+            if (authorInList != null)
```

for simplicity let's throw an exception when adding an author with an existing name. that way our method will truly add a new author when it's successful.

---

Пока не делал, может всетаки оставим?:
1) да я согласен (в плане), что возмжно оно незначительно улучшит читаемость кода (см. 3). Но так мы ухудшим функциональность;
2) Допустим, мы добавляем книгу с автором, который уже есть. Вместо того, чтобы добавить новую книгу, а автора взять из базы - мы вызовем искючение (показвая, что у нас что-то не так = хотя у автора может быть не одна книга, а несколько) 
3) возможно улучшить читаемость данного метода, это вынести поиск автора в отдельный метод, возвращающего автора, оъединить returnAuthor с authorInList... незнаю как улучшить читаемость метода
4) Но считаю, что вызывать исключение там, где условно метод всетаки должен работать -это не вариант. Хотя улучшение читаемости не мешало: с ходу если посмотреть, трудно читаеться...